BARS-20 Создать страницы регистрации и авторизации

Helmoz opened the pull request.

Helmoz committed:
BARS-20 Добавил email в claims jwt токена

Добавил метод для проверки занятости Email

Helmoz committed:
BARS-20 Реализованы страницы регистрации и авторизации пользователя

Валидация для формы, сохранение токена после входа

Helmoz committed:
BARS-20 Реализован вывод сообщения об ошибке при авторизации пользователя

Добавлены кнопки перехода между страницами авторизации и регистрации

d4n0n-myself commented on Platform/Platform.Domain/DomainServices/UserDomainService.cs:
```diff
@@ -24,7 +24,14 @@ public OperationResult CheckUserExitence(string login)
             var user = _repository.FindByPredicate(x => x.Login == login);
 
-            return new OperationResult(user != null);
+            return new OperationResult(user == null);
```
это ты специально так поменял ?

Helmoz replied:
Да, нужно поменять названия методов, проверяем занят ли login/email

d4n0n-myself commented on Platform/Platform.Domain/Services/TokenService.cs:
```diff
@@ -12,26 +12,27 @@ namespace Platform.Domain.Services
 	public class TokenService
 	{
-		internal JwtSecurityToken GenerateToken(string login)
+		internal JwtSecurityToken GenerateToken(string login, string email)
```
мб стоит просто модель User пробросить? в месте создания токена по идее всегда есть модель, и дополнительную не придется создавать при создании IUserIdentity

Helmoz replied:
Сделано

d4n0n-myself commented on Platform/Platform.Web/ClientApp/src/models/modules/jwt-decode.d.ts:
```diff
@@ -0,0 +1 @@
+declare module 'vue-jwt-decode'
```
?

Helmoz replied:
Так надо

d4n0n-myself commented on Platform/Platform.Web/ClientApp/src/pages/authentication/Login.vue:
```diff
@@ -0,0 +1,70 @@
+    <q-card square style="width: 400px; padding:50px">
+      <q-card-section>
+        <div class="text-h6">
+          Авторизация
```
мб просто Вход?

Helmoz replied:
Можно попробовать

Helmoz replied:
Сделано

d4n0n-myself commented on Platform/Platform.Web/ClientApp/src/pages/authentication/Login.vue:
```diff
@@ -0,0 +1,70 @@
+      </q-card-section>
+      <q-card-actions class="justify-center">
+        <q-btn color="primary" class="full-width" @click="login">
+          Авторизоваться
```
Войти?

Helmoz replied:
Ща попробую

Helmoz replied:
Сделано

d4n0n-myself commented on Platform/Platform.Web/ClientApp/src/pages/authentication/Register.vue:
```diff
@@ -0,0 +1,221 @@
+            id="email"
+            v-model.trim="email"
+            type="email"
+            label="Email"
```
Почтовый адрес/почта? Надо уж весь интерфейс по-русски писать

Helmoz replied:
Тут не согласен, Email вроде как и в русской локализации норм отображается

Helmoz replied:
Решено переименовать -> Почта

d4n0n-myself commented on Platform/Platform.Web/ClientApp/src/pages/authentication/Register.vue:
```diff
@@ -0,0 +1,221 @@
+      return `Это поле обязательное для заполнения`
+    }
+    if (val.hasOwnProperty('email') && !val.email) {
+      return `Введите корректный Email`
```
тоже перевести на рус

Helmoz replied:
Сделано

d4n0n-myself commented on Platform/Platform.Web/Controllers/UsersController.cs:
```diff
@@ -27,12 +27,23 @@ public IActionResult CheckUserExistence(string login)
+        /// Used to log in and receive new JWT.
+        /// </summary>
+        /// <returns>Json with 1 property: "token" : *token*</returns>
+        [HttpGet]
```
Отступ можешь поправить?

d4n0n-myself replied:
Ctrl + Alt + Enter достаточно

Helmoz replied:
++

d4n0n-myself commented on Platform/Platform.Domain/Services/TokenService.cs:
```diff
@@ -12,26 +12,27 @@ namespace Platform.Domain.Services
 			{
-				new Claim(ClaimTypes.Name, user.Login)
+				new Claim(ClaimTypes.Name, user.Login),
+                new Claim(ClaimTypes.Email, user.Email)
```
Ctrl + Alt + Enter (отступ )

Helmoz replied:
++

d4n0n-myself reviewed (changes requested):
Пара косметических исправлений, и пару вопросов / мб улучшений

Helmoz committed:
BARS-20 Исправление по ревью

При формировании токена прокидывается модель user'a, а не логин с почтой
Добавлена валидация при входе(required поля)
Правки codestyle и локализации

d4n0n-myself committed:
testing

d4n0n-myself committed:
отступы

d4n0n-myself commented on Platform/Platform.Domain/Services/TokenService.cs:
```diff
@@ -12,30 +13,30 @@ namespace Platform.Domain.Services
+		private ClaimsIdentity GetIdentity(User user)
 		{
-			var user = new {Login = login, Email = email, Role = new {Name = "admin"}}; // get user by login from db
+            var userWithRole = new { user.Login, user.Email, Role = new {Name = "admin"}}; //add role to User model?
```
такие комменты желательно в todo'шки бросать

d4n0n-myself committed:
Merge branch 'dev' into feature/BARS-20
